File: src/main/java/ast/ASTIfStmnt.java
package ast;

public class ASTIfStmnt extends ASTNode {
    public ASTExpr cond;
    public ASTNodeArray<ASTNode> code;

    public ASTIfStmnt(ASTFileLocation loc, ASTExpr cond, ASTNodeArray<ASTNode> code){
        super(loc);
        this.cond = cond;
        this.code = code;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTIfStmnt: (cond, code)");
        cond.print(i+1);
        code.print(i+1);
    }
}

File: src/main/kotlin/ast/ASTIfStmnt.kt
package ast

class ASTIfStmnt(loc: ASTFileLocation, val cond: ASTExpr, val code: ASTNodeArray<ASTNode>) : ASTNode(loc), ASTBlockStmnt {

    override fun print(i: Int) {
        printIndent(i)
        println("ASTIfStmnt: (cond, code)")
        cond.print(i + 1)
        code.print(i + 1)
    }

    override fun getBlock(): ASTNodeArray<ASTNode> {
        return code
    }
}